modules/twinklewarn.js

```diff
@@ -1452,115 +1268,130 @@ Twinkle.warn.callback.change_category = function twinklewarnCallbackChangeCatego
 			}
-			e.target.form.block_timer.disabled = true;
-			e.target.form.block_timer.value = 'indefinite';
-		} else if( e.target.form.block_timer.disabled ) {
+			$(form.block_timer).prop('readonly', true);
+			form.block_timer.value = 'indefinite';
+		} else if( $(form.block_timer).prop('readonly') ) {
 			if(Twinkle.warn.prev_block_timer !== null) {
-				e.target.form.block_timer.value = Twinkle.warn.prev_block_timer;
+				form.block_timer.value = Twinkle.warn.prev_block_timer;
 				Twinkle.warn.prev_block_timer = null;
 			}
-			e.target.form.block_timer.disabled = false;
+			$(form.block_timer).prop('readonly', false);
 		}
 
-		if( Twinkle.warn.messages.block[value].pageParam ) {
+		if( Twinkle.block.blockPresetsInfo[value].pageParam ) {
 			if(Twinkle.warn.prev_article !== null) {
-				e.target.form.article.value = Twinkle.warn.prev_article;
+				form.article.value = Twinkle.warn.prev_article;
 				Twinkle.warn.prev_article = null;
 			}
-			e.target.form.article.disabled = false;
-		} else if( !e.target.form.article.disabled ) {
+			$(form.article).prop('readonly', false);
+		} else if( !$(form.article).prop('readonly') ) {
 			if(Twinkle.warn.prev_article === null) {
-				Twinkle.warn.prev_article = e.target.form.article.value;
+				Twinkle.warn.prev_article = form.article.value;
+			}
+			$(form.article).prop('readonly', true);
+			form.article.value = '';
+		}
+
+		if( Twinkle.block.blockPresetsInfo[value].noDurationParam ) {
+			if(Twinkle.warn.prev_block_timer === null) {
+				Twinkle.warn.prev_block_timer = form.block_timer.value || '';
 			}
-			e.target.form.article.disabled = true;
-			e.target.form.article.value = '';
+			$(form.block_timer).prop('readonly', true);
+			form.block_timer.value = '';
+		} else if( $(form.block_timer).prop('readonly') && !Twinkle.block.blockPresetsInfo[value].indefinite ) {
+			if(Twinkle.warn.prev_block_timer !== null) {
+				form.block_timer.value = Twinkle.warn.prev_block_timer;
+				Twinkle.warn.prev_block_timer = null;
+			}
+			$(form.block_timer).prop('readonly', false);
 		}
 
-		if( Twinkle.warn.messages.block[value].reasonParam ) {
+		if( Twinkle.block.blockPresetsInfo[value].reasonParam ) {
 			if(Twinkle.warn.prev_block_reason !== null) {
-				e.target.form.block_reason.value = Twinkle.warn.prev_block_reason;
+				form.block_reason.value = Twinkle.warn.prev_block_reason;
 				Twinkle.warn.prev_block_reason = null;
 			}
-			e.target.form.block_reason.disabled = false;
-		} else if( !e.target.form.block_reason.disabled ) {
+			$(form.block_reason).prop('readonly', false);
+		} else if( !form.block_reason.readonly ) {
 			if(Twinkle.warn.prev_block_reason === null) {
-				Twinkle.warn.prev_block_reason = e.target.form.block_reason.value;
+				Twinkle.warn.prev_block_reason = form.block_reason.value;
 			}
-			e.target.form.block_reason.disabled = true;
-			e.target.form.block_reason.value = '';
+			$(form.block_reason).prop('readonly', true);
```

Comment:
I haven't tested this yet, but with readonly, don't you lose the UI affordance (i.e. graying out of the form field)?

In any case, isn't it a moot point, since we're not actually "submitting" the form to a GET or POST endpoint (a la plain old HTML), but we're simply reading off the values in JS?